packages/core/src/tools/facades/BlockToolFacade.ts

```diff
@@ -0,0 +1,217 @@
+import type { BlockToolAdapter } from "@editorjs/dom-adapters";
+import { BaseToolFacade, InternalBlockToolSettings, UserSettings } from './BaseToolFacade.js';
+import type {
+  BlockAPI,
+  BlockToolData,
+  ConversionConfig,
+  PasteConfig, SanitizerConfig, ToolboxConfig,
+  ToolboxConfigEntry
+} from '@editorjs/editorjs';
+import { isEmpty, cacheable, isObject } from '@editorjs/helpers';
+import { type InlineToolFacade } from './InlineToolFacade.js';
+import { ToolType } from './ToolType.js';
+import { type BlockTuneFacade } from './BlockTuneFacade.js';
+import { ToolsCollection } from './ToolsCollection.js';
+import { BlockToolConstructor as BlockToolConstructable, BlockTool as IBlockTool } from '../../entities/BlockTool.js';
+
+/**
+ * Class to work with Block tools constructables
+ */
+export class BlockToolFacade extends BaseToolFacade<ToolType.Block, IBlockTool> {
+  /**
+   * Tool type — Block
+   */
+  public type: ToolType.Block = ToolType.Block;
+
+  /**
+   * InlineTool collection for current Block Tool
+   */
+  public inlineTools: ToolsCollection<InlineToolFacade> = new ToolsCollection<InlineToolFacade>();
+
+  /**
+   * BlockTune collection for current Block Tool
+   */
+  public tunes: ToolsCollection<BlockTuneFacade> = new ToolsCollection<BlockTuneFacade>();
+
+  /**
+   * Tool's constructable blueprint
+   */
+  protected declare constructable: BlockToolConstructable;
+
+  /**
+   * Creates new Tool instance
+   * @param options
+   * @param options.data - Tool data
+   * @param options.block - BlockAPI for current Block
+   * @param options.readOnly - True if Editor is in read-only mode
+   */
+  public create({ data, block, readOnly, adapter }: { data: BlockToolData, block: BlockAPI, readOnly: boolean, adapter: BlockToolAdapter }): IBlockTool {
+    return new this.constructable({
+      adapter,
+      data,
+      block,
+      readOnly,
+      api: this.api,
+      config: this.settings,
+    });
+  }
+
+  /**
+   * Returns true if read-only mode is supported by Tool
+   */
+  public get isReadOnlySupported(): boolean {
+    return this.constructable[InternalBlockToolSettings.IsReadOnlySupported] === true;
+  }
+
+  /**
+   * Returns true if Tool supports linebreaks
+   * @todo check if we still need this as BlockToolConstructable doesn't have line breaks option
```

Comment:
Code tool [uses](https://github.com/editor-js/code/blob/master/src/index.ts#L89) it.

Reply:
We don't have it in the types, so we'll need to update types first